### PR TITLE
DYN-6927: Improvements to RemovePIIData API 

### PR DESCRIPTION
**Pull request by reddyashish**
### Purpose

Task: https://jira.autodesk.com/browse/DYN-6927

Using the RemovePIIData(JObject jsonObject), we check for the regex patterns in specific parts like the VIew or Annotations section of the JSON to remove the personal sensitive data.

It should remove email pattern, SSN number pattern, credit card pattern, user directory pattern, IP address pattern, date patterns and website patterns.

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes
DYN-6927: Improvements to RemovePIIData API to accurately remove only the sensitive personal info.

### Reviewers
@QilongTang @zeusongit 

### FYIs
@DynamoDS/dynamo

### CODE REVIEW

File: src/DynamoMLDataPipeline/DynamoMLDataPipeline.cs [hunk 1]
```diff
@@ -161,7 +161,16 @@ static public string ConvertDynToBase64(string filePath)
...
+
+            JObject jsonObject = JObject.Parse(sourceFileContent);
+
+            Tuple<JObject, bool> jsonResult = PIIDetector.RemovePIIData(jsonObject);
+
+            string formattedSourceContent = jsonObject.ToString();
+            if (jsonResult.Item2)
+            {
+                formattedSourceContent = jsonResult.Item1.ToString();
+            }
```
Comment (QilongTang) [inline src/DynamoMLDataPipeline/DynamoMLDataPipeline.cs:173]: Would you put some comments on these changes?

File: src/DynamoMLDataPipeline/DynamoMLDataPipeline.cs [hunk 2]
```diff
@@ -161,7 +161,20 @@ static public string ConvertDynToBase64(string filePath)
         {
             // Read .dyn file as a string
             string sourceFileContent = File.ReadAllText(filePath);
-            string formattedSourceContent = PIIDetector.RemovePIIData(sourceFileContent);
+
+            //Parsing the workspace info into a JSON object.
+            JObject jsonObject = JObject.Parse(sourceFileContent);
+
+            // RemovePIIData will remove the sensitive data from specific parts of JSON structure(like the Views, Annotiations and code block info)
+            Tuple<JObject, bool> jsonResult = PIIDetector.RemovePIIData(jsonObject);
```
Comment (QilongTang) [inline src/DynamoMLDataPipeline/DynamoMLDataPipeline.cs:169]: Maybe string node as well?

**Review by QilongTang — APPROVED** [review]
One comment then LGTM

File: src/DynamoUtilities/PIIDetector.cs [hunk 3]
```diff
@@ -119,15 +119,13 @@ internal static string RemovePIIData(string data)
...
             result = Regex.Replace(data, emailPattern, "");
             result = Regex.Replace(result, directoryPattern, "");
             result = Regex.Replace(result, ssnPattern, "");
             result = Regex.Replace(result, datePattern, "");
-            //result = Regex.Replace(result, websitePattern, "");
-            //result = Regex.Replace(result, creditCardPattern, "");
-            //result = Regex.Replace(result, ipPattern, "");
+            result = Regex.Replace(result, ipPattern, "");
+            result = Regex.Replace(result, creditCardPattern, "");
+            result = Regex.Replace(result, websitePattern, "");
```
Comment (QilongTang) [inline src/DynamoUtilities/PIIDetector.cs:128]: hi @reddyashish So the trick is to just change the order of regex?

Comment (reddyashish) [reply src/DynamoUtilities/PIIDetector.cs:128]: The main change is to use RemovePIIData(JObject jsonObject) API on the whole json object which will look for the regex pattern in the Views, Annotations etc.

Comment (QilongTang) [reply src/DynamoUtilities/PIIDetector.cs:128]: Right, in the past we are running the regex against the text?

Comment (reddyashish) [reply src/DynamoUtilities/PIIDetector.cs:128]: yes

**Pull request merged**